src/Nowruz/modules/dispute/components/contributorDashboard/useContributorDashboard.tsx

```diff
@@ -0,0 +1,151 @@
+import { Cell, ColumnDef, getCoreRowModel, useReactTable } from '@tanstack/react-table';
+import { useEffect, useMemo, useState } from 'react';
+import variables from 'src/components/_exports.module.scss';
+import { Dispute } from 'src/core/api';
+import { disputes, LeaveContribution } from 'src/core/api/disputes/disputes.api';
+import { toRelativeTime } from 'src/core/relative-time';
+import { Chip } from 'src/Nowruz/modules/general/components/Chip';
+import { Dot } from 'src/Nowruz/modules/general/components/dot';
+
+export const useContributorDashboard = (setJoined: (val: boolean) => void) => {
+  const [stopNotif, setStopNotif] = useState(false);
+  const [list, setList] = useState<Dispute[]>([]);
+  const [openModal, setOpenModal] = useState(false);
+
+  useEffect(() => {
+    const getDisputes = async () => {
+      const res = await disputes({ page: 1, limit: 5 });
+      setList(res.items);
+    };
+    getDisputes();
```

Comment:
There is a single route: `/:id/contribute` for both dashboard and contributor-join-program
if user is already joined, dashboard is loading, otherwise contributor-join-program page is loading
since it is conditionally rendered, data is not fetched in preload of page